intermediate-exercises/src/components/03-FetchGithubData.js

```diff
@@ -23,41 +23,66 @@ import React, { Component } from 'react';
 const GithubRepos = ({ repos }) => {
   return (
     <ul>
-      {/* Task: The list of repos here */}
+      {repos.map(({ id, name }) => (
+        <li key={id}>{name}</li>
+      ))}
     </ul>
   );
-}
+};
 
 // Task: Open the console in the browser. There will be a warning
 // about incorrect prop type for user.
 // Define the correct prop type for the prop `repos`
 GithubRepos.propTypes = {
-
+  repos: PropTypes.array.isRequired
 };
 
 /* eslint-disable react/no-multi-comp */
 class UsernameForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      username: '',
-      repos: [],
+      username: "",
+      repos: []
     };
   }
+  getUserRepos = async () => {
+    const { username } = this.state;
+
+    if (!username || username.trim() === "") {
+      console.log(
+        "Please provide a username to continue fetching the repositories"
+      );
+      return;
+    }
+
+    try {
+      const { data: repos } = await axios.get(
+        `https://api.github.com/users/${username}/repos`
+      );
+      this.setState({ repos: repos });
+    } catch (error) {
+      console.log("error", error);
```

Comment:
I think error message can be more descriptive

Reply:
Thanks totally agreed. I will look into it.